### PR TITLE
Add support for Arm based images in OCI

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
What this PR does / why we need it:
This PR has the necessary changes to build Arm based images on Oracle Cloud Infrastructure.

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

**Additional context**
Add any other context for the reviewers

### CODE REVIEW

File: images/capi/ansible/roles/node/meta/main.yml [hunk 1]
```diff
@@ -19,8 +19,14 @@ dependencies:
       debs: "{{ common_debs }}"
     when: ansible_distribution == "Amazon"
 
+  - role: setup
+    vars:
+      rpms: "{{ common_rpms }}"
+      debs: "{{ common_debs }}"
+    when: packer_builder_type == "oracle-oci" and ansible_architecture == "aarch64"
```
Comment (shyamradhakrishnan) [inline images/capi/ansible/roles/node/meta/main.yml:26]: Arm does not have the common virt rpms and debs

File: images/capi/ansible/roles/setup/tasks/debian.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@
     src: etc/apt/sources.list.j2
     dest: /etc/apt/sources.list
     mode: 0644
+  when: packer_builder_type != "oracle-oci"
```
Comment (shyamradhakrishnan) [inline images/capi/ansible/roles/setup/tasks/debian.yml:20]: OCI Base images has the required apt sources list embedded inside the image. IF we copy the templated file, Arm based build are failing as the sources list is not correct for Arm

**Comment by codenrhoden** [comment]
overall the changes look fine to me.
/lgtm

I suspect there is some overall refactoring to be done to properly allow a choice of CPU arch -- x86 is pretty much hardcoded across the project for all providers. So I think other providers would run into similar issues around trying to reach `common_virt` packages that don't exist on ARM.

File: images/capi/ansible/roles/setup/tasks/debian.yml (hunk 2 above)
Comment (jsturtevant) [reply images/capi/ansible/roles/setup/tasks/debian.yml:20]: This seems like a good bit of information to have in a comment so it is not lost in future

**Comment by codenrhoden** [comment]
/approve
@shyamradhakrishnan has mentioned in Slack that he's got a follow-on PR in progress where he can add the requested comment.

**Comment by jsturtevant** [comment]
azure flake was https://github.com/kubernetes-sigs/image-builder/issues/766.  

ova failed on another PR today to (#825). Not sure if there is something flaky or if it is configuration for ova builds:

```
vsphere-clone: fatal: [default]: UNREACHABLE! => {"changed": false, "msg": "Failed to create temporary directory.In some cases, you may have been able to authenticate and did not have permissions on the target directory.
```

/retest

File: images/capi/ansible/roles/setup/tasks/debian.yml (hunk 2 above)
Comment (shyamradhakrishnan) [reply images/capi/ansible/roles/setup/tasks/debian.yml:22]: Added comment

**Pull request merged**